pkgs/development/python-modules/ruff-lsp/default.nix

```
@@ -0,0 +1,43 @@
+{ lib
+, pythonOlder
+, buildPythonPackage
+, fetchPypi
+, ruff
+, pygls
+, hatchling
+, typing-extensions
+}:
+
+buildPythonPackage rec {
+  pname = "ruff-lsp";
+  version = "0.0.17";
+  format = "pyproject";
+  disabled = pythonOlder "3.7";
+
+  src = fetchPypi {
+    inherit version;
+    pname = "ruff_lsp";
+    sha256 = "sha256-5xKIhLz06y1CxLhtJpUOhcoB7HOi5mFEKINcgZs4dsU=";
+  };
+
+  nativeBuildInputs = [
+    hatchling
+  ];
+
+  propagatedBuildInputs = [
+    pygls
+    typing-extensions
+  ];
+
+  postPatch = ''
+    sed -i '/"ruff>=/d' pyproject.toml
+    sed -i 's|USER_DEFAULTS: dict\[str, str\] = {}|USER_DEFAULTS: dict[str, str] = {"path": ["${ruff}/bin/ruff"]}|' ruff_lsp/server.py
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/charliermarsh/ruff-lsp";
+    description = "A Language Server Protocol implementation for Ruff";
```

The changelog will be automatically attached to the commit messages by tool (e.g., `nix-update`).

```suggestion
    description = "A Language Server Protocol implementation for Ruff";
    changelog = "https://github.com/charliermarsh/ruff-lsp/releases/tag/v${version}";
```